runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts

```diff
@@ -2,22 +2,212 @@ import { Injectable } from '@nestjs/common';
+
+        const workbookSessionInfo: WorkbookSessionInfo = await this.microsoftGraphService.post(
+            this.createWorkbookUrl(request),
+            {
+                persistChanges: persistChanges,
+            }
+        );
+
+        this.session = {
+            ...this.session,
+            workbookSessionInfo,
+        };
+
+        return workbookSessionInfo;
+    }
+    //https://learn.microsoft.com/en-us/graph/api/workbook-closesession?view=graph-rest-1.0&tabs=http
+    public async closeSession(): Promise<void> {
+        const request = '/closeSession';
+
+        return this.microsoftGraphService.post(
+            this.createWorkbookUrl(request),
+            {},
+            {
+                headers: {
+                    'workbook-session-id': this.session.workbookSessionInfo.id,
+                },
+            }
+        );
+    }
+    // https://learn.microsoft.com/en-us/graph/api/worksheet-cell?view=graph-rest-1.0&tabs=http
+    public async getCell(cellCoordinates: WorkbookCellCoordinates): Promise<WorkbookRange> {
+        const request = `/worksheets/${this.session.worksheetIdentifier}/cell(row=${
+            Number(cellCoordinates.row) - 1
+        },column=${this.getColumnNumber(cellCoordinates.column) - 1})`;
+
+        return this.microsoftGraphService.get(this.createWorkbookUrl(request), {
+            headers: {
+                'workbook-session-id': this.session.workbookSessionInfo.id,
+            },
+        });
+    }
+
+    // https://learn.microsoft.com/en-us/graph/api/worksheet-range?view=graph-rest-1.0&tabs=http
+    public async getRange(address: string): Promise<WorkbookRange> {
+        const request = `/worksheets/${this.session.worksheetIdentifier}/range(address='${address}')`;
+
+        return this.microsoftGraphService.get(this.createWorkbookUrl(request), {
+            headers: {
+                'workbook-session-id': this.session.workbookSessionInfo.id,
+            },
+        });
+    }
+
+    // https://learn.microsoft.com/en-us/graph/api/range-insert?view=graph-rest-1.0&tabs=http
+    public async setCell(address: string, value: string): Promise<WorkbookRange> {
+        const request = `/worksheets/${this.session.worksheetIdentifier}/range(address='${address}:${address}')`;
+
+        const newRange: WorkbookRangeUpdateBody = {
+            values: [[value]],
+        };
+
+        return this.microsoftGraphService.patch(this.createWorkbookUrl(request), newRange, {
+            headers: {
+                'workbook-session-id': this.session.workbookSessionInfo.id,
+            },
+        });
+    }
+
+    // https://learn.microsoft.com/en-us/graph/api/range-insert?view=graph-rest-1.0&tabs=http
+    public async setRange(address: string, values: string | any[][]) {
+        const request = `/worksheets/${this.session.worksheetIdentifier}/range(address='${address}')`;
+
+        if (!Array.isArray(values)) {
+            values = JSON.parse(values) as any[][];
```

Comment:
use unknown instead of any but here imo `(string | number)[][]` is more appriopriate